core/src/main/scala/magnolia1/monadic.scala

```diff
@@ -0,0 +1,56 @@
+package magnolia1
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Success, Try}
+
+trait Functor[F[_]] {
```

Comment:
do we need the `Functor` as a stand-alone abstraction anywhere? My worry would be that it's such a common name, that we would add to the import-pollution when magnolia is on the classpath. Maybe we could simply have only `Monadic` with all the methods?